src/service/github.oauth.service.ts

```diff
@@ -0,0 +1,71 @@
+import * as express from "express";
+import { Server } from "http";
+import fetch from "node-fetch";
+import { URLSearchParams } from "url";
+import { state } from "../state";
+
+export class GitHubOAuthService {
```

Comment:
how are we handling exceptions in this entire class ?
We need to throw manual exceptions like other chunks in repo.

Reply:
If an error occurs during the `fetch` to GitHub (most likely because of invalid URL), `Commons.LogException` is called with the error and a possible cause ("Sync: Invalid GitHub Enterprise URL."). Otherwise, the error is logged with `Commons.LogException` using the generic error message, informing the user to check the console.